src/Application/Middleware/AssetsMiddleware.php

```diff
@@ -0,0 +1,104 @@
+<?php
+declare(strict_types=1);
+
+namespace Externals\Application\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Stratify\Http\Middleware\Middleware;
+use Zend\Diactoros\Response;
+use Zend\Diactoros\Stream;
+
+/**
+ * Serve assets.
+ *
+ * @author Matthieu Napoli <matthieu@mnapoli.fr>
+ */
+class AssetsMiddleware implements Middleware
+{
+    /**
+     * @var string
+     */
+    private $publicPath;
+
+    public function __construct(string $publicPath)
+    {
+        $this->publicPath = rtrim($publicPath, '/') . '/';
+    }
+
+    public function __invoke(ServerRequestInterface $request, callable $next) : ResponseInterface
+    {
+        $path = $request->getUri()->getPath();
+
+        $file = $this->publicPath . $path;
```

Comment:
It's probably best not to implement your own, e.g. Aerys has that built-in. Not sure whether ReactPHP's server has something like that yet.